infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -221,6 +221,7 @@ public void write(CorfuPayloadMsg<WriteRequest> msg, ChannelHandlerContext ctx,
 
         batchWriter.addTask(WRITE, msg)
                 .thenRunAsync(() -> {
+                    logData.serializeMetadata();
```

Comment:
This should be in the cache put function. Otherwise it is a bug.

Reply:
Done, thanks!